source/pdo_sqlsrv/pdo_dbh.cpp

```diff
@@ -1343,7 +1343,7 @@ int pdo_sqlsrv_dbh_quote( pdo_dbh_t* dbh, const char* unquoted, size_t unquoted_
         }
         // get the placeholder at the current position in driver_stmt->placeholders ht
         zval* placeholder = NULL;
-        if (( placeholder = zend_hash_get_current_data( driver_stmt->placeholders )) != NULL && zend_hash_move_forward( driver_stmt->placeholders ) == SUCCESS ) {
+        if (( placeholder = zend_hash_get_current_data( driver_stmt->placeholders )) != NULL && zend_hash_move_forward( driver_stmt->placeholders ) == SUCCESS && stmt->bound_params != NULL ) {
```

Comment:
decided not to change the ordering because "placeholder = zend_hash_get_current_data( driver_stmt->placeholders )) != NULL" is more likely to fail and if it fails the if statement can just move on without doing other checks